examples/99-advanced/planets.py

```diff
@@ -0,0 +1,119 @@
+
+Creates high-resolution renderings of the Earth and the major celestial bodies
+in our solar system for astrodynamics applications.
+This example is inspired by `planet3D-MATLAB <https://github.com/tamaskis/planet3D-MATLAB>`_ .
+
+
+Note that this example author doesn't know anything about astronomy. There may
+be inaccuracies in the representation. The purpose of this example is for
+inspiration only.
+
+"""
+import pyvista
+from pyvista import examples
+
+###############################################################################
+# Celestial bodies with the Milky Way in the background.
+
+
+# Light of the sun in the +x axis direction.
+
+light = pyvista.Light()
+light.set_direction_angle(30, -20)
+
+# Load planet
+
+# https://tamaskis.github.io/files/Visualizing_Celestial_Bodies_in_3D.pdf
+# Mercury's radius is 2439.0 km
+mercury = examples.planets.load_mercury(radius=2439.0, lat_resolution=150, lon_resolution=300)
+# Venus's radius is 6052.0 km
+venus = examples.planets.load_venus(radius=6052.0, lat_resolution=150, lon_resolution=300)
+# Mars's radius is 3397.2 km
+mars = examples.planets.load_mars(radius=3397.2, lat_resolution=150, lon_resolution=300)
+# Jupiter's radius is 71492.0 km
+jupiter = examples.planets.load_jupiter(radius=71492.0, lat_resolution=150, lon_resolution=300)
+# Saturn's radius is 60268.0 km
+saturn = examples.planets.load_saturn(radius=60268.0, lat_resolution=150, lon_resolution=300)
+# Saturn's rings range from 7000.0 km to 80000.0 km from the surface of the planet
+inner = 60268.0 + 7000.0
+outer = 60268.0 + 80000.0
+saturn_ring_alpha = examples.planets.load_saturn_ring_alpha(inner=inner, outer=outer, c_res=50)
+# Uranus's radius is 25559.0 km
+uranus = examples.planets.load_uranus(radius=25559.0, lat_resolution=150, lon_resolution=300)
+# Neptune's radius is 24764.0 km
+neptune = examples.planets.load_neptune(radius=24764.0, lat_resolution=150, lon_resolution=300)
+
+# Move planet position (Numbers have no meaning. The planets are laid out for
+# easy viewing.).
+
+mercury.translate((0.0, 0.0, 0.0), inplace=True)
+venus.translate((-15000.0, 0.0, 0.0), inplace=True)
+mars.translate((-45000.0, 0.0, 0.0), inplace=True)
+jupiter.translate((-150000.0, 0.0, 0.0), inplace=True)
+saturn.translate((-400000.0, 0.0, 0.0), inplace=True)
+saturn_ring_alpha.translate((-400000.0, 0.0, 0.0), inplace=True)
+uranus.translate((-600000.0, 0.0, 0.0), inplace=True)
+neptune.translate((-700000.0, 0.0, 0.0), inplace=True)
+
+# Add planets to Plotter.
+
+plotter = pyvista.Plotter(lighting="none")
+cubemap = examples.download_cubemap_space_16k()
+_ = plotter.add_actor(cubemap.to_skybox())
+plotter.set_environment_texture(cubemap, True)
+plotter.add_light(light)
+plotter.add_mesh(mercury)
+plotter.add_mesh(venus)
+plotter.add_mesh(mars)
+plotter.add_mesh(jupiter)
+plotter.add_mesh(saturn)
+plotter.add_mesh(saturn_ring_alpha)
+plotter.add_mesh(uranus)
+plotter.add_mesh(neptune)
+plotter.camera.zoom(1.5)
+plotter.show()
```

Comment:
I'd define `smooth_shading=True` through a theme. As it is now, if you zoom in on a planet you can see the grid:
![Screenshot from 2022-08-20 16-07-34](https://user-images.githubusercontent.com/17914410/185750271-4820a19c-6371-4f3f-b092-7ebe0b300cda.png)

Later this is also directly visible in the last (Venus) figure on the left:
![sphx_glr_planets_003](https://user-images.githubusercontent.com/17914410/185750712-66cfee64-c5ed-488a-8cdd-96dedf44b0d3.png)

This will probably go away if we enable smooth shading.

Reply:
```suggestion
plotter = pyvista.Plotter(lighting="none")
cubemap = examples.download_cubemap_space_16k()
_ = plotter.add_actor(cubemap.to_skybox())
plotter.set_environment_texture(cubemap, True)
plotter.add_light(light)
plotter.add_mesh(mercury, smooth_shading=True)
plotter.add_mesh(venus, smooth_shading=True)
plotter.add_mesh(mars, smooth_shading=True)
plotter.add_mesh(jupiter, smooth_shading=True)
plotter.add_mesh(saturn, smooth_shading=True)
plotter.add_mesh(saturn_ring_alpha, smooth_shading=True)
plotter.add_mesh(uranus, smooth_shading=True)
plotter.add_mesh(neptune, smooth_shading=True)
plotter.add_mesh(pluto,  smooth_shading=True)
plotter.camera.zoom(1.5)
plotter.show()
```